### PR TITLE
Added RecoveryServices.Backup module

### PR DESCRIPTION
**Pull request by dragonfly91**
Corresponding Swagger PRs are merged: 
https://github.com/Azure/azure-rest-api-specs/pull/584
https://github.com/Azure/azure-rest-api-specs/pull/625

### CODE REVIEW

**Comment by markcowl** [comment]
@dragonfly91 We cannot merge this PR until the Swagger change has been merged.  Please update the title when that occurs.

**Comment by markcowl** [comment]
@dragonfly91 Please clean up the commits - there should not be merge commits ot multiple commits with the same comment in the history.  You cna follow this advice: https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md for cleaning up commits.

**Review by markcowl — CHANGES_REQUESTED** [review]
@dragonfly91 Please clean up the commits - there should not be merge commits ot multiple commits with the same comment in the history. You cna follow this advice: https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md for cleaning up commits.

**Comment by dragonfly91** [comment]
@markcowl I've simplified the commits. Can you continue with the review?

File: src/ResourceManagement/RecoveryServices.Backup/Microsoft.Azure.Management.RecoveryServices.Backup/Customizations/Client.Customizations.cs
Comment (markcowl) [inline src/ResourceManagement/RecoveryServices.Backup/Microsoft.Azure.Management.RecoveryServices.Backup/Customizations/Client.Customizations.cs]: project.json
'- This version should be a preview.  Please append -prview to the version in Newtonsoft.Json
- Consider wither this should be a new library with new namespaces and a new name,m or if it shoudl use the same name as the previous library.


generate.cmd
- broken link to your swagger spec.  Your link should us e a particular commit, and not a link to the branch, like this:
https://raw.githubusercontent.com/Azure/azure-rest-api-specs/188c597d6e87fede4a3b0048ddff3fbfdb7cfd4c/arm-recoveryservicesbackup/compositeRecoveryServicesBackupClient.json

Tests - Your tests are dependnt on pre-created artifacts.  You should have all necessary setup in the test setup.  If this is not possible for some reason, then you need to describe in detail how to create the Azure structures necessary to run the test
SceanrioTests.cs
-= it looks like you are setting the test fixture in each test run.  Please follow the pattern - per-test setup can happen in the test class constructor, per suite setup should happen in a test fixture.

**Review by markcowl — CHANGES_REQUESTED** [review]

File: src/ResourceManagement/RecoveryServices.Backup/Microsoft.Azure.Management.RecoveryServices.Backup/Customizations/Client.Customizations.cs
Comment (dragonfly91) [inline src/ResourceManagement/RecoveryServices.Backup/Microsoft.Azure.Management.RecoveryServices.Backup/Customizations/Client.Customizations.cs]: Regarding the comment about generate.cmd, can you give more clarity about which commit should it be? Also, we've tried to follow the Compute project as they've just specified the link to the composite JSON spec: https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/generate.cmd

Why is this guideline different?

File: src/ResourceManagement/RecoveryServices.Backup/RecoveryServices.Backup.Tests/RecoveryServicesBackupTestBase.cs
Comment (markcowl) [inline src/ResourceManagement/RecoveryServices.Backup/RecoveryServices.Backup.Tests/RecoveryServicesBackupTestBase.cs]: This is not what I meant.  See the discussion of class and collection fixtures here: https://xunit.github.io/docs/shared-context.html

File: src/ResourceManagement/RecoveryServices.Backup/RecoveryServices.Backup.Tests/TestHelpers/BackupDefinition.cs
Comment (markcowl) [inline src/ResourceManagement/RecoveryServices.Backup/RecoveryServices.Backup.Tests/TestHelpers/BackupDefinition.cs]: This is not sufficient.  You are using particular names for these resources, so you need to specify what they are, or provide a mechanism for a user to provide resource names and details without changing the code.

**Review by markcowl — COMMENTED** [review]
Required changes in text fixture per xunit practices for per test / per class / per collection text fixtures.  Also, please augment your instructions on creation

File: src/ResourceManagement/RecoveryServices.Backup/RecoveryServices.Backup.Tests/RecoveryServicesBackupTestBase.cs
Comment (dragonfly91) [inline src/ResourceManagement/RecoveryServices.Backup/RecoveryServices.Backup.Tests/RecoveryServicesBackupTestBase.cs]: We've followed the per-test setup pattern (first model) as suggested in the link.

File: src/ResourceManagement/RecoveryServices.Backup/RecoveryServices.Backup.Tests/TestHelpers/BackupDefinition.cs
Comment (dragonfly91) [inline src/ResourceManagement/RecoveryServices.Backup/RecoveryServices.Backup.Tests/TestHelpers/BackupDefinition.cs]: I've moved the test artifact related values to a json file and have included this detail in the comment.

**Comment by dragonfly91** [comment]
Addressed the comments, please review / merge!

**Comment by markcowl** [comment]
@dragonfly91 Still need a correct reference to your swagger document

**Comment by dragonfly91** [comment]
Addressed the comments, please review / merge!

**Review by markcowl — APPROVED** [review]

**Pull request merged**
